README.md

```diff
@@ -60,7 +60,12 @@ Once you have implemented a lower bound search for the start of the range, imple
 
 *How do you use binary search to find the lower bound of a number? How did you have to modify the binary search algorithm?*
 
+We search with binary splits to iterate outward from the middle value to find value v. If the middle value x[mid] != v then we iterate the middle value 1 towards v, and set this middle value to high or low depending on whether x[mid] is higher than or lower than v. We repeat this until we hit the value v with the binary splits, and then we keep moving our lower bound such that our high and low value will meet at the first v value found in the sorted list.
+ 
 *Would anything be more difficult if the features covered ranges instead of single nucleotides (like real BED files)? What could go wrong, if anything?*
 
+Yes it would make it more difficult. It would mean that the present comparison of only chrom index and chrom start would not nr sufficient for sorting the chromosomes. Overlaps in chrom 'start' and 'end' interval would occur. If we wanted a completely sorted and merged outfile we would have to take both chrom 'start' and chrom 'end' into account in the merging.
+
 *We wrote a tool for merging two BED files, but what if we had a bunch of them? What would the complexity be if we merged them in, one at a time? What would the complexity be if we merged all of the files at the same time?*
 
+If we merge them one by one and we describe number of bedlines as m and the length of the bedlines as n, then we compare each bedline with eachother, and iterate through the lenght to merge them such that when compared one by one we get O(n*m). If comparing all the bedlines at once then we still have to compare all the bedlines at once such that we still compare m bedlines iterating over n, giving the same O(n*m)
```

Comment:
The final output would indeed have size $nm$, but producing it is not quite that fast.

If you merge one at a time, the first merge would take time $2m$ (you merge two lists of size $m$ each). The second would take $3m$, the third $4m$, and so on, so the time would be $m(2+3+4+\cdots+n-1)\in O(mn^2)$.

Merging all the files at once would require that you find the smallest elements out of the current $n$ lists in each iteration. If you do that with a linear search, you spend $O(n)$ each time you output a single element, and then we are again at $O(mn^2)$. You can speed that up, using a data structure we haven't seen yet, so it becomes $O(\log n)$ to find and update the smallest element in each iteration, and then you are at $O(mn\log n)$.

A divide and conquer approach could be to merge two halfs of the input files first, and then merge the results. That would be the recurrence $T(n,m)=2T(n/2,m)+O(n\cdot m)$ which would also take you to $O(mn\log n)$.